src/lib/transaction_snark/test/zkapps_examples/big_circuit/big_circuit.ml

```diff
@@ -0,0 +1,42 @@
+open Transaction_snark_tests.Util
+open Core_kernel
+open Mina_base
+open Signature_lib
+module Impl = Pickles.Impls.Step
+module Inner_curve = Snark_params.Tick.Inner_curve
+module Nat = Pickles_types.Nat
+module Local_state = Mina_state.Local_state
+module Zkapp_command_segment = Transaction_snark.Zkapp_command_segment
+
+let sk = Private_key.create ()
+
+let pk = Public_key.of_private_key_exn sk
+
+let pk_compressed = Public_key.compress pk
+
+let account_id = Account_id.create pk_compressed Token_id.default
+
+(* we want to create a circuit with a domain of size 2^16 *)
+let num_constraints = 1 lsl 16
+
+let expected_err =
```

Comment:
I think we want a successful proof with `2^16` constraints. This isn't exercising pickles verification.
You'll probably need to set `num_constraints = 1 lsl 15` to account for pickles overhead.

Reply:
do you mean that we need two tests: one with 2^16 constraints that work, and one with 2^16+X constraints that doesn't? I'm a bit confused here